engine/src/main/java/org/terasology/logic/players/FirstPersonHeldItemMountPointComponent.java

```diff
@@ -85,13 +84,8 @@ public void poseChanged(Matrix4f pose, int handIndex) {
             return;
         }
         trackingDataReceived = true;
-        Matrix4f adjustedPose = pose.mul(toolAdjustmentMatrix);
-        translate = new Vector3f(adjustedPose.m30(), adjustedPose.m31(), adjustedPose.m32());
-        org.joml.Vector4f jomlQuaternion = org.terasology.rendering.openvrprovider.OpenVRUtil.convertToQuaternion(adjustedPose);
-        if (rotationQuaternion == null) {
-            rotationQuaternion = new Quat4f(jomlQuaternion.x, jomlQuaternion.y, jomlQuaternion.z, jomlQuaternion.w);
-        } else {
-            rotationQuaternion.set(jomlQuaternion.x, jomlQuaternion.y, jomlQuaternion.z, jomlQuaternion.w);
-        }
+        Matrix4f adjustedPose = pose.mul(toolAdjustmentMatrix, new Matrix4f());
```

Comment:
I made the null check obsolete by making the members final. The client shouldn't be able to reset or set them to null anyway. The rest is simplification.